rclpy/src/rclpy/_rclpy.c

```diff
@@ -2073,25 +2082,11 @@ rclpy_destroy_entity(PyObject * Py_UNUSED(self), PyObject * args)
     PyMem_Free(node);
   } else if (PyCapsule_IsValid(pyentity, "rcl_timer_t")) {
     rcl_timer_t * timer = (rcl_timer_t *)PyCapsule_GetPointer(pyentity, "rcl_timer_t");
-    rcl_clock_t * clock = 0;
-    ret = rcl_timer_clock(timer, &clock);
-    if (ret != RCL_RET_OK) {
-      PyErr_Format(PyExc_RuntimeError,
-        "Failed to get 'rcl_clock_t' from timer: %s", rcl_get_error_string_safe());
-      rcl_reset_error();
-      return NULL;
-    }
-    rcl_ret_t ret_clock = rcl_clock_fini(clock);
-    if (ret_clock != RCL_RET_OK) {
-      PyErr_Format(PyExc_RuntimeError,
-        "Failed to fini 'rcl_clock_t': %s", rcl_get_error_string_safe());
-      rcl_reset_error();
-      return NULL;
-    }
-
     ret = rcl_timer_fini(timer);
     PyMem_Free(timer);
-    PyMem_Free(clock);
+  } else if (PyCapsule_IsValid(pyentity, "rcl_clock_t")) {
+    PyCapsule_SetDestructor(pyentity, NULL);
+    _rclpy_destroy_clock(pyentity);
```

Comment:
The pycapsule destructor will fini the clock if the user doesn't. If the clock is destroyed explicitly then fini the clock, but set the destructor to NULL so the garbage collector doesn't do it again.